plumbing/msg/sender.go

```diff
@@ -20,17 +20,27 @@ import (
 // Topic is the network pubsub topic identifier on which new messages are announced.
 const Topic = "/fil/msgs"
 
+// Abstracts over a store of blockchain state.
+type chainState interface {
+	// LatestState returns the latest chain state.
+	LatestState(ctx context.Context) (state.Tree, error)
+}
+
 // PublishFunc is a function the Sender calls to publish a message to the network.
 type PublishFunc func(topic string, data []byte) error
 
 // Sender is plumbing implementation that knows how to send a message.
 type Sender struct {
 	// Signs messages.
 	signer types.Signer
-	// Provides actor state (we could reduce this dependency to only LatestState()).
-	chainReader chain.ReadStore
-	// Pool of existing message, and receiver of the sent message.
-	msgPool *core.MessagePool
+	// Provides actor state
+	chainState chainState
+	// Provides the current block height
+	blockTimer core.BlockTimer
```

Comment:
@acruikshank in a follow-up I suggest removing the blocktimer from message pool in favour of this one.

While the chain store ends up providing both `chainState` and `core.blockTimer` it's very useful in testing that they're separate interfaces here.